avocado_vt/plugins/vt_joblock.py

```diff
@@ -0,0 +1,119 @@
+import errno
+import logging
+import os
+import re
+import random
+import string
+import sys
+
+from avocado.core import exit_codes
+from avocado.core.settings import settings
+from avocado.plugins.base import JobPre, JobPost
+
+from ..test import VirtTest
+
+
+class VTJobLock(JobPre, JobPost):
+
+    name = 'vt-joblock'
+    description = 'Avocado-VT Job Lock/Unlock'
+
+    def __init__(self):
+        self.log = logging.getLogger("avocado.app")
+        self.has_vt_test = False
+        self.lock_dir = settings.get_value(section="plugins.vtjoblock",
+                                           key="dir",
+                                           key_type=str,
+                                           default='/tmp')
```

Comment:
It never came to my mind, but how about using `data_dir.get_data_dir()/avocado-vt/lock-$uid-$pid`?

1. data_dir is always read-write
2. the $uid might come handy in case datadir is in /tmp
3. we're not checking other data_dirs, so simultaneous executions with different datadir is possible

We should create the files as 666, try removing the ones associated with dead pid and ignore exceptions in case the lock belonged to other user (we might not have permissions to remove it when datadir is in /tmp).

@clebergnu what do you think about this? I think that should work, simplify things significantly and even allow non-corruptible parallel executions.